src/transaction.js

```diff
@@ -125,11 +125,18 @@ function Transaction(session, options) {
    */
   this.transaction = true;
 
+  options = options || {};
+
   this.queuedMutations_ = [];
   this.runFn_ = null;
 
-  this.timeout_ = DEFAULT_TRANSACTION_TIMEOUT;
   this.beginTime_ = null;
+  this.timeout_ = DEFAULT_TRANSACTION_TIMEOUT;
+
+  if (options.timeout) {
```

Comment:
Will anyone ever want `0`?

Reply:
I don't think so, but I've added a `is.number()` just to be safe.